tests/test_routes.py

```diff
@@ -0,0 +1,346 @@
+import contextlib
+import logging
+from nose.tools import (
+    assert_raises,
+    eq_,
+    set_trace,
+)
+import flask
+from flask import Response
+from werkzeug.exceptions import MethodNotAllowed
+
+from core.app_server import ErrorHandler
+from core.opds import VerboseAnnotator
+
+from app import app
+from controller import MetadataWrangler
+from problem_details import INVALID_CREDENTIALS
+import routes
+
+from test_controller import ControllerTest
+
+class MockMetadataWrangler(object):
```

Comment:
This code is very similar to [the mock controller scaffolding in circulation](https://github.com/NYPL-Simplified/circulation/blob/master/tests/test_routes.py#L33), but it's not the same because here the application and authentication setup are different (simpler).